### PR TITLE
(#164) Improve repositories management

### LINKED ISSUE #164: Package repositories management is convoluted
The `choria::repo` class has a few issues:

1. On Debian, packages are always taken from the "stretch" releases, regardless of the current distro.  However, buster packages are available;
2. Nightly packages configuration only does something on yum repositories, while the project now also provides debian nightly packages (even if they will be dropped in the short term, `choria::nightly_repo` doing nothing on Debian is unexpected);
3. Overall, the code has some duplication.

### PR DESCRIPTION
**Pull request by smortex**
This is a collection of minor enhancements for the `choria::repo` class.

Fixes #164

### CODE REVIEW

File: manifests/repo.pp [hunk 1]
```diff
@@ -6,6 +6,10 @@
   Enum["present", "absent"] $ensure = "present",
 ) {
   assert_private()
+  $nightly_ensure = $nightly ? {
+    true  => 'present',
```
Comment (ripienaar) [inline manifests/repo.pp:10]: please use double quotes everywhere

File: manifests/repo.pp [hunk 2]
```diff
@@ -32,33 +36,29 @@
...
+    if ! $release in ["xenial", "bionic", "stretch", "buster"] {
+      fail("Choria Repositories are not supported on ${release}")
     }
+
     apt::source{"choria-release":
       ensure        => $ensure,
       notify_update => true,
       comment       => "Choria Orchestrator Releases",
-      location      => "${choria::repo_baseurl}/release/ubuntu/",
+      location      => "${choria::repo_baseurl}/release/${facts["os"]["name"].downcase}/",
```
Comment (ripienaar) [inline manifests/repo.pp:61]: with these awful quoting that the hashes cause I prefer `sprintf("%s/release/%s/", $choria::repo_baseurl, $facts["os"]["name"].downcase)` this avoid double quotes in double quotes

File: manifests/repo.pp (hunk 1 above)
Comment (smortex) [reply manifests/repo.pp]: Fixed!

File: manifests/repo.pp (hunk 2 above)
Comment (smortex) [reply manifests/repo.pp]: Done!

File: manifests/repo.pp [hunk 3]
```diff
@@ -67,21 +67,21 @@
...
-      ensure        => $ensure,
+
+    apt::source{"choria-nightly":
+      ensure        => $nightly_ensure,
       notify_update => true,
-      comment       => "Choria Orchestrator Releases",
-      location      => "${choria::repo_baseurl}/release/debian/",
-      release       => "stretch",
+      comment       => "Choria Orchestrator Nightly Builds",
+      location      => "${choria::repo_baseurl}/nightly/${facts["os"]["name"].downcase}/",
```
Comment (ripienaar) [inline manifests/repo.pp:75]: ah sorry, I didnt see these nested `"` before, also make sprintf please

Comment (smortex) [reply manifests/repo.pp]: Ah, I fixed the previous issues in full automatic mode, before coffee (and waking up completely obviously)…  Did not think about the nightly repo that mimics the release one.

**Pull request merged**